### PR TITLE
Added an optional {-o/--output-file} argument to oesign sign.

### LINKED ISSUE #3281: Add -o option to oesign sign
`oesign sign` currently doesn't accept an option to specify the output filename / path. Could this be added? It would make certain flows in a build system less painful.

### PR DESCRIPTION
**Pull request by RRathna**
Fixes  #3281.

### CODE REVIEW

File: tools/oesign/main.c [hunk 1]
```diff
@@ -60,6 +61,8 @@ static const char _usage_sign[] =
     "                           certificate used to sign the digest file.\n"
     "  -d, --digest-file        path to the signed digest file matching the\n"
     "                           enclave image and specified configuration.\n"
+    "  -o, --output-file        [OPTIONAL] file name (with path) where\n"
```
Comment (mingweishih) [inline tools/oesign/main.c:64]: For consistency:

```suggestion
    "  -o, --output-file        [optional] file name (with path) where the\n"
```

File: tools/oesign/main.c [hunk 2]
```diff
@@ -60,6 +61,8 @@ static const char _usage_sign[] =
...
+    "                           signature of enclave image will be written.\n"
```
Comment (mingweishih) [inline tools/oesign/main.c:65]:
```suggestion
    "                           signature of the enclave image will be written.\n"
```

**Review by mingweishih — APPROVED** [review]
LGTM with minor comments.

File: tools/oesign/main.c [hunk 3]
```diff
@@ -60,6 +61,9 @@ static const char _usage_sign[] =
     "                           certificate used to sign the digest file.\n"
     "  -d, --digest-file        path to the signed digest file matching the\n"
     "                           enclave image and specified configuration.\n"
+    "  -o, --output-file        [OPTIONAL] file name (with path) where the\n"
```
Comment (mingweishih) [inline tools/oesign/main.c:64]: We were using `optional` (lowercase) in other options. It would be great to keep the consistency :)

Comment (RRathna) [reply tools/oesign/main.c]: Ah I missed it, I'll do that :-)

**Review by anakrish — APPROVED** [review]
LGTM

This change is worth a note in the CHANGELOG since it affects the user experience.

**Pull request merged**